File: backend/internal/domain/entities/exchange_rate.go
package entities

import "time"

type ExchangeRate struct {
	ID        string     `json:"id"`
	Rate      float64    `json:"rate"`
	ValidFrom time.Time  `json:"valid_from"`
	ValidTo   *time.Time `json:"valid_to,omitempty"`
	CreatedAt time.Time  `json:"created_at"`
}

File: frontend/lib/modules/exchange_rate/domain/entities/exchange_rate.dart
class ExchangeRate {
  const ExchangeRate({
    required this.id,
    required this.rate,
    required this.validFrom,
    required this.createdAt,
    this.validTo,
  });

  final String id;
  final double rate;
  final DateTime validFrom;
  final DateTime? validTo;
  final DateTime createdAt;
}
